src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -2240,14 +2239,33 @@ public void migrationMailArchiveZip(HttpServletRequest request,
 			HttpServletResponse response) {
 
 		// zip download
-		HashMap<String, String> callStatus = migration_call(request, response,
-				Utils.MIGRATION_MAILARCHIVE_TYPE_ZIP, Utils.getCurrentUserId(request, env));
-		if (callStatus.containsKey("errorMessage")) {
-			log.info(this + " migrationMailArchiveZip call error message="
-					+ callStatus.get("errorMessage"));
-		} else if (callStatus.containsKey("migrationId")) {
-			log.info(this + " migrationMailArchiveZip call migration started id="
-					+ callStatus.get("migrationId"));
-		}
+        Map<String, String[]> parameterMap = request.getParameterMap();
+        String destination_type = parameterMap.get("destination_type")[0];
+
+        if(destination_type.equals("mailarchivezip")) {
+            HashMap<String, String> callStatus = migration_call(request, response,
```

Comment:
No need to repeat this code.  Extract a single method to make call and do logging.  Pass in the Utils.MIGRATION_MAILARCHIVE_TYPE_.... and the string name of the migration type to put in the log message.